src/multi.c

```diff
@@ -191,6 +193,14 @@ void execCommand(client *c) {
         c->argv_len = c->mstate.commands[j].argv_len;
         c->cmd = c->mstate.commands[j].cmd;
 
+        /* Note that if refcount == 1 means that the module of this
+         * command has been unloaded elsewhere, we need to skip it. */
+        if (c->cmd->refcount == 1) {
+            addReplyErrorFormat(c, "Invalid command: %s, the module of this command has been unloaded",
+                                c->cmd->fullname);
```

Comment:
Instead of relying on refcount == 1, can MODULE UNLOAD set `cmd->func` to NULL? Then exec doesn't need to check anything. Only RedisModuleCommandDispatcher() needs to check `cmd->func` before calling it and return the error.

Reply:
Good idea.

Reply:
This does not seem to work, after `moduleFreeCommand()`, this module command should not be used again except for `cmd->declared_name` and `cmd->fullname`, but it will continue to be used in `call()`.

Reply:
OK, I think it doesn't work because things like keyspecs are already freed, but the pointers in cmd are not cleared.

But maybe `refcount == 1` is not always correct. If some other client has this command as its lastcmd, then refcount > 1 here.

So, either we can check something else here, like `if (c->cmd->flags & CMD_MODULE && c->cmd->func == NULL)` instead of refcount, or we can move the call to moduleFreeCommand to decrCommandRefCount, so that almost everything in cmd (such as keyspecs) still exists so ACL check etc. works as before. WDYT?

Reply:
Thanks, please take a look, I moved the moduleFreeCommand to decrCommandRefCount to free the module command and it seems to work fine,